File: OOP based implmentation/OOP based implmentation/Project/login/StudentSignup.java
package login;
import objs.user;

public class StudentSignup extends user {
    private String rollNum;
    private String course;

    public StudentSignup(String rollNum, String course, String username, String password) {
        super(username, password, "student");
        this.rollNum = rollNum;
        this.course = course;

    }
   
    public String getRollNum()
    {
        return rollNum;
    }
    public String getCourse()
    {
        return course;
    }
}

File: OOP based implmentation/OOP based implmentation/Project/login/StudentSignup.cpp
#include <string>
#include "StudentSignup.h"

using namespace std;

StudentSignup::StudentSignup(const string &rollNum, const string &course, const string &username, const string &password)
    : User(username, password, "student"), rollNum(rollNum), course(course)
{
}

string StudentSignup::getRollNum() const
{
    return rollNum;
}

string StudentSignup::getCourse() const
{
    return course;
}